dowhy/causal_identifier/id_identifier.py

```diff
@@ -97,102 +96,107 @@ def __init__(self, graph, estimand_type, method_name="default", proceed_when_uni
         :param method_name: Identification method ("id-algorithm" in this case).
         :param proceed_when_unidentifiable: If True, proceed with identification even in the presence of unobserved/missing variables.
         """
-
-        super().__init__(graph, estimand_type, method_name, proceed_when_unidentifiable)
-
-        if self.estimand_type != CausalIdentifier.NONPARAMETRIC_ATE:
+        if estimand_type != CausalIdentifierEstimandType.NONPARAMETRIC_ATE:
             raise Exception("The estimand type should be 'non-parametric ate' for the ID method type.")
 
-        self._treatment_names = OrderedSet(parse_state(graph.treatment_name))
-        self._outcome_names = OrderedSet(parse_state(graph.outcome_name))
-        self._adjacency_matrix = graph.get_adjacency_matrix()
-
-        try:
-            self._tsort_node_names = OrderedSet(
-                list(nx.topological_sort(graph._graph))
-            )  # topological sorting of graph nodes
-        except:
-            raise Exception("The graph must be a directed acyclic graph (DAG).")
-        self._node_names = OrderedSet(graph._graph.nodes)
-
-    def identify_effect(self, treatment_names=None, outcome_names=None, adjacency_matrix=None, node_names=None):
+    def identify_effect(self, graph: CausalGraph, treatment_name, outcome_name, node_names=None, **kwargs):
```

Comment:
Can we add some mypy types here to treatment name, outcome_name and node_names? Does kwargs need to be documented in the PyDoc section below?

Reply:
**kwargs is only here to keep the method signature between different identifiers the same to be able to use the `CausalIdentifier` protocol. I'll add documentation on the Protocol class and also the identify_effect on both the Backdoor and ID Identifiers.